include/session/config/groups/members.hpp

```diff
@@ -155,20 +168,55 @@ struct member {
 
     /// API: groups/member::set_promoted
     ///
-    /// Sets the "promoted" flag for this user.  This marks the user as having a pending
-    /// promotion-to-admin in the group.  The optional `failed` parameter can be specified as true
-    /// if the promotion was issued but failed to send for some reason (this is intended as a signal
-    /// to other clients that the promotion should be reissued).
+    /// This marks the user as having a pending promotion-to-admin in the group, waiting for the
+    /// promotion message to be sent to them.
+    void set_promoted() {
+        admin = true;
+        invite_status = 0;
+        promotion_status = INVITE_NOT_SENT;
+    }
+
+    /// API: groups/member::set_promotion_sent
     ///
-    /// Note that this flag is ignored when the `admin` field is set to true.
+    /// This marks the user as having a pending promotion-to-admin in the group, and that a
+    /// promotion message has been sent to them.
+    void set_promotion_sent() {
+        admin = true;
+        invite_status = 0;
+        promotion_status = INVITE_SENT;
+    }
+
+    /// API: groups/member::set_promotion_failed
     ///
-    /// Inputs:
-    /// - `failed`: can be specified as true to mark the promotion status as "failed-to-send".  If
-    ///   omitted or false then the promotion status is set to "sent".
-    void set_promoted(bool failed = false) {
-        promotion_status = failed ? INVITE_FAILED : INVITE_SENT;
+    /// This marks the user as being promoted to an admin, but that their promotion message failed
+    /// to send (this is intended as a signal to other clients that the promotion should be
+    /// reissued).
+    void set_promotion_failed() {
+        admin = true;
+        invite_status = 0;
+        promotion_status = INVITE_FAILED;
+    }
+
+    /// API: groups/member::accept_promotion
+    ///
+    /// This marks the user as having accepted a promotion to admin in the group.
+    void accept_promotion() {
+        admin = true;
+        invite_status = 0;
+        promotion_status = 0;
     }
 
+    /// API: groups/member::promotion_needs_send
+    ///
+    /// Returns whether the user needs a promotion to admin status to be sent.
+    /// Returns true if so (whether or not that promotion has failed).
+    ///
+    /// Inputs: None
+    ///
+    /// Outputs:
+    /// - `bool` -- true if the user needs a promotion to be sent, false otherwise.
+    bool promotion_needs_send() const { return promotion_status == INVITE_NOT_SENT; }
```

Comment:
same comment, but this one might be a bit off with the right meaning too as this method wouldn't return true of `promotion_status == INVITE_FAILED` which is what `set_promotion_failed` does

Reply:
Yea, I think this one is actually a bit more misleading than the other one...

Actually, looking at the logic I've changed the behaviour of the `admin` flag from "accepted admin" to be "someone that has been promoted to admin" - I'll update the functions accordingly (I think this behaviour makes more sense as we should consider a member an admin as soon as the promotion process has started, rather than only if they've confirmed receipt of the admin keys)